examples/example-gcp-csm-observability/src/main/java/io/grpc/examples/csmobservability/CsmObservabilityClient.java

```diff
@@ -0,0 +1,149 @@
+   * greeting. The second argument is the target server.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "world";
+    String target = "xds:///helloworld:50051";
+    int prometheusPort = 9465;
+    if (args.length > 0) {
+      if ("--help".equals(args[0])) {
+        System.err.println("Usage: [name [target [prometheusPort]]]");
+        System.err.println("");
+        System.err.println("  name    The name you wish to be greeted by. Defaults to " + user);
+        System.err.println("  target  The server to connect to. Defaults to " + target);
+        System.err.println("  prometheusPort  The port to expose prometheus metrics. Defaults to " + prometheusPort);
+        System.exit(1);
+      }
+      user = args[0];
+    }
+    if (args.length > 1) {
+      target = args[1];
+    }
+    if (args.length > 2) {
+      prometheusPort = Integer.parseInt(args[2]);
+    }
+
+    Thread mainThread = Thread.currentThread();
+
+    Runtime.getRuntime().addShutdownHook(new Thread() {
+      @Override
+      public void run() {
+        // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+        System.err.println("*** shutting down gRPC client since JVM is shutting down");
+
+        sendRpcs = false;
+        try {
+          mainThread.join();
+        } catch (InterruptedException e) {
+          e.printStackTrace(System.err);
+        }
+        System.err.println("*** client shut down");
+      }
+    });
+
+    /**
+     * Adds a PrometheusHttpServer to convert OpenTelemetry metrics to Prometheus format and
+     * expose these via a HttpServer exporter to the SdkMeterProvider.
+     */
+    SdkMeterProvider sdkMeterProvider = SdkMeterProvider.builder()
+        .registerMetricReader(
+            PrometheusHttpServer.builder().setPort(prometheusPort).build())
+        .build();
+
+    /** Initialize OpenTelemetry SDK with MeterProvider configured with Prometeheus. */
+    OpenTelemetrySdk openTelemetrySdk =
+        OpenTelemetrySdk.builder().setMeterProvider(sdkMeterProvider).build();
+
+    /** Initialize CSM Observability. */
+    CsmObservability observability = CsmObservability.newBuilder()
+        .sdk(openTelemetrySdk)
+        .build();
+    /** Registers CSM observabiity globally. */
+    observability.registerGlobal();
+
+    /** Create a communication channel to the server, known as a Channel. */
+    ManagedChannel channel = Grpc.newChannelBuilder(target, InsecureChannelCredentials.create())
+        .build();
+
+    /** Run RPCs every second. */
+    while (sendRpcs) {
+      CsmObservabilityClient client = new CsmObservabilityClient(channel);
+      client.greet(user);
+      // Sleep for a bit before sending the next RPC.
+      Thread.sleep(3000);
+    }
+
+    /** Close CSM Observability. */
+    if (observability != null) {
+      observability.close();
+    }
+    openTelemetrySdk.close();
+    channel.shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
```

Comment:
Let's shutdown this channel before observability, so that we shut down in the opposite order of creation.

Reply:
Done